code/game/objects/items/weapons/RCD.dm

```diff
@@ -382,3 +389,285 @@ RCD
 	name = "admin RCD"
 	max_matter = INFINITY
 	matter = INFINITY
+
+
+/obj/item/weapon/rcd/arcd
+	name = "advanced rapid-construction-device (ARCD)"
+	max_matter = 300
+	matter = 300
+	color = "red"
+
+
+// Ranged RCD
+
+
+/obj/item/weapon/rcd/arcd/afterattack(atom/A, mob/user, proximity)
+	proximity = 1
```

Comment:
In base RCD: add something like `var/ranged = FALSE` and set it to true for your ranged RCDs, then stick it in the proximity check.